### PR TITLE
Web Request additions

### PR DESCRIPTION
**Pull request by TcT2k**
* Add `wxWebRequest::DisablePeerVerify()` to allow connection to insecure SSL hosts
* Some documentation additions

### CODE REVIEW

File: include/wx/private/webrequest.h [hunk 1]
```diff
@@ -96,6 +96,10 @@ class wxWebRequestImpl : public wxRefCounterMT
 
     virtual wxWebRequestHandle GetNativeHandle() const = 0;
 
+    void DisablePeerVerify(bool disable) { m_peerVerifyDisabled = disable; }
+
+    bool IsPeerVerifyDisabled() { return m_peerVerifyDisabled; }
```
Comment (vadz) [inline include/wx/private/webrequest.h:101]: Really minor, but:
```suggestion
    bool IsPeerVerifyDisabled() const { return m_peerVerifyDisabled; }
```

File: include/wx/webrequest.h [hunk 2]
```diff
@@ -188,6 +188,10 @@ class WXDLLIMPEXP_NET wxWebRequest
 
     wxWebRequestHandle GetNativeHandle() const;
 
+    void DisablePeerVerify(bool disable = true);
+
+    bool IsPeerVerifyDisabled();
```
Comment (vadz) [inline include/wx/webrequest.h:193]: Somewhat more important here:
```suggestion
    bool IsPeerVerifyDisabled() const;
```

File: interface/wx/protocol/http.h [hunk 3]
```diff
@@ -12,6 +12,9 @@
 
     wxHTTP can thus be used to create a (basic) HTTP @b client.
 
+    @note If you want HTTPS, IPv6, Proxy detection, authentication, etc.
+        support consider using wxWebRequest instead.
```
Comment (vadz) [inline interface/wx/protocol/http.h:16]: I'd be slightly more forceful and say
```suggestion
    @note In practice, for any but the most trivial cases, e.g. if you need HTTPS, HTTP/2 or IPv6,
            proxy detection, authentication, etc. support please use wxWebRequest instead.
```

File: interface/wx/webrequest.h [hunk 4]
```diff
@@ -391,6 +399,22 @@ class wxWebRequest
...
+        insecure.
+    */
+    void DisablePeerVerify(bool disable = true);
+
+    /**
+        Returns if peer verification has been disabled.
+
+        @see DisablePeerVerify()
+    */
+    bool IsPeerVerifyDisabled();
```
Comment (vadz) [inline interface/wx/webrequest.h:417]:
```suggestion
    bool IsPeerVerifyDisabled() const;
```

**Review by vadz — APPROVED** [review]
Thanks a lot! I can merge this as is and/or apply my suggestions myself if you agree (or at least don't disagree).

**Comment by TcT2k** [comment]
I've applied your suggestions

**Comment by vadz** [comment]
We're probably going to have to disable the tests on AppVeyor, the network just seems too flaky there :-(

**Comment by TcT2k** [comment]
> We're probably going to have to disable the tests on AppVeyor, the network just seems too flaky there :-(

I agree, it seems to unreliable. Some things could be tested by running https://httpbin.org/ locally, but I'm not sure if it's worth the effort.

**Comment by vadz** [comment]
It still seems suspicious to me that in 9 cases out of 10 it's the `Cancel` test which is failing -- yet I can never reproduce this locally, even after running it a hundred times. I wonder what's going on here...

**Pull request merged**